src/utils/Block.ts

```diff
@@ -0,0 +1,225 @@
+import { EventBus } from "./EventBus";
+import { nanoid } from 'nanoid';
+
+// Нельзя создавать экземпляр данного класса
+class Block<P extends Record<string, any> = any> {
```

Comment:
тут можно избежать использования any
```suggestion
class Block<P extends Record<string, unknown> = Record<string, never>> {
```
таким образом мы указываем дефолтное значение для типа пропсов, чтобы не указывать пропсы в тех компонентах, в которых их нет (как например в Mainlogo)